rust/pact_ffi/src/mock_server/handles.rs

```diff
@@ -2227,6 +2231,66 @@ ffi_fn!{
   }
 }
 
+ffi_fn!{
+  /// Add a text comment to the interaction.
+  ///
+  /// * `interaction` - Interaction handle to set the comments for.
+  /// * `comment` - Comment value.
+  ///
+  /// This function will return `true` if the comments were successfully
+  /// updated. The `comment` must be a valid UTF-8 null-terminated string.
+  ///
+  /// Unlike [`pactffi_set_comment`], this function will always append the
+  /// comment to the array of comments under the `text` key.
+  ///
+  /// If, for any reason, the `text` key is not present or the associated
+  /// value not an array, it will be created as/replaced by an array and the
+  /// comment will be appended to it.
+  ///
+  /// # Safety
+  ///
+  /// The comments parameter must be a valid pointer to a NULL terminated UTF-8,
+  /// or NULL if the comment is to be cleared.
+  fn pactffi_add_text_comment(interaction: InteractionHandle, comment: *const c_char) -> bool {
+    let comment = match convert_cstr("comment", comment) {
+      Some(comment) => comment,
+      None => {
+        error!("add_text_comment: Comment value is not valid (NULL or non-UTF-8)");
+        return Err(anyhow!("Comment value is not valid (NULL or non-UTF-8)"));
+      }
+    };
+
+    let ensure_append = |comments: &mut serde_json::Value| {
+      match comments {
+        serde_json::Value::Array(_) => {
+          comments.as_array_mut().unwrap().push(serde_json::Value::String(comment.to_string()));
```

Comment:
You've already pattern matched on the array, you can just use the inner Vec and not need to call as_array and unwrap